### PR TITLE
Refactor all

### PR DESCRIPTION
**Pull request by doktordirk**

### CODE REVIEW

**Comment by AdamWillden** [comment]
I've always thought the "current" property was a bit weird but never actually questioned it. 

@RWOverdijk may know of a reason the property was required but if there's no reason I'm all for this change.

Much clearer and less weirdness!

**Comment by AdamWillden** [comment]
I just thumbed up my own message. Oh how stupid I look.

**Comment by RWOverdijk** [comment]
@AdamWillden I never used current. I would never even call it current. It was just named current when we started using it.

File: package.json [hunk 1]
```diff
@@ -47,15 +49,16 @@
     }
   },
   "dependencies": {
+    "aurelia-api": "^2.1.2",
     "aurelia-dependency-injection": "^1.0.0-beta.1.2.0",
     "aurelia-fetch-client": "^1.0.0-beta.1.2.0",
     "aurelia-router": "^1.0.0-beta.1.2.0",
-    "aurelia-api": "^2.1.2"
+    "extend": "^3.0.0"
```
Comment (RWOverdijk) [inline package.json:59]: Do we really have to define our deps in threefold?

File: src/app.fetch-httpClient.config.js [hunk 2]
```diff
@@ -70,34 +70,34 @@ export class FetchConfig {
...
 
   /**
-   * @param {HttpClient|Rest[]} client
+   * @param {HttpClient|Rest[]} aClient
    *
    * @return {HttpClient[]}
    */
-  configure(client) {
-    if (Array.isArray(client)) {
+  configure(aClient) {
```
Comment (RWOverdijk) [inline src/app.fetch-httpClient.config.js]: Why aClient?

Comment (doktordirk) [inline src/app.fetch-httpClient.config.js]: client is a property already or something. i hate it if i can't search.

File: package.json (hunk 1 above)
Comment (doktordirk) [inline package.json:59]: dont get it

File: src/app.fetch-httpClient.config.js (hunk 2 above)
Comment (RWOverdijk) [inline src/app.fetch-httpClient.config.js]: I prefer something like httpClient then. "aClient" is a bit weird imo.

**Comment by doktordirk** [comment]
@AdamWillden i refactoring it all further. that might invalidate your plans there. maybe but your plans into an issue and i get them in with reference

**Comment by doktordirk** [comment]
preview

**Comment by AdamWillden** [comment]
Hmm I replaced `getCurrentToken` with `getAccessToken`. I'm sure I ran the tests before committing....

**Comment by AdamWillden** [comment]
Ah right I didn't change it. You did :-D its in your commit not mine.

**Comment by doktordirk** [comment]
@RWOverdijk @AdamWillden i've done what i wanted mostly. one thing is that 
  accessTokenProp  accessTokenName,  accessTokenRoot can be combined into one dot delimited parameter without loss. wys?

else please review the general stuff (no need for formating etc yet).

meanwhile i add more tests

ps: should we get some file naming convention in place?

**Comment by AdamWillden** [comment]
> one thing is that accessTokenProp accessTokenName, accessTokenRoot can be combined into one dot delimited parameter without loss. wys?

I'm still game for this.

What do you mean by "without loss"? We won't lose functionality if that's what you mean. It would of course be a breaking change (which I know you know).

> ps: should we get some file naming convention in place?

Such as?

> else please review the general stuff (no need for formating etc yet).

Just a note in the future it'd be better for review purposes if that commit were broken down into a few commits. You seem to do a lot of unrelated changes in single commits. That's just my preference though. @RWOverdijk  may prefer things wrapped up into a single commit. I don't know.

---

You should remove the first commit in this PR as I remove the method/function altogether in.. oh no I don't remove it but I should have. Okay, 2 things then:
1. Remove your first commit
2. Amend my first commit to remove the `addTokenPrefix` function

**Comment by doktordirk** [comment]
re: breaking up commits. right, but in this case all files are changed. it's better to just look at the current version instead of the changes

**Comment by doktordirk** [comment]
finished first pass. working again. i'll see if i can make it BC

**Pull request merged**